Obfuscar/Obfuscator.cs

```diff
@@ -148,47 +148,49 @@ public void RunRules()
         internal HashSet<string> CollectTypesFromXaml()
         {
             var xamlFiles = Project.AssemblyList.SelectMany(info => GetXamlDocuments(info.Definition));
-            var namesInXaml = NamesInXaml(xamlFiles);
+            var typeNamesInXaml = TypeNamesInXaml(xamlFiles).ToHashSet();
 
+            void AddToNamesInXamlIfRequired(TypeReference type)
+            {
+                var typeDefinition = type.Resolve();
+                if (typeDefinition != null && typeDefinition.IsEnum)

```

Comment:
Currently we are trying to handle all in this collect method.
New implementation does not change this but adds a method more to detect all types in relation to an implementation of `INotifyPropertyChange`.

I actually wanted to remove the enum types not related to the BAML collected types and only use that for properties as it probably should have been done like that in the first place, but I do not want to break any backward combability.

Also worth mentioning: We only use types from BAML extraction - we could look into `PropertyRecord`s to combine that with public/internal properties in types in relation to an implementation of `INotifyPropertyChange`. This could probably give us a few more percent from things not currently used in bindings but do support it.

This I also chose not to pursue as we have lots of manual notified properties which use strings or other logic which could be broken by such changes.

Lots of room for improvement for the future but lets take it in steps.